### PR TITLE
Components restructuring

### PR DESCRIPTION
**Pull request by vladcipariu91**
## Description

Updated the structure of `appyx-components` to:

- eperimental
   - cards
   - promoter
   - puzzle15    
- internal
   - test-drive
- stable
   - backstack
   - spotlight  

## Check list

- [ ] I have updated `CHANGELOG.md` if required.
- [ ] I have updated documentation if required.

### CODE REVIEW

File: appyx-components/stable/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/stable/backstack/BackStack.kt [hunk 1]
```diff
@@ -1,8 +1,8 @@
-package com.bumble.appyx.components.backstack
+package com.bumble.appyx.components.stable.backstack
```
Comment (KovalevAndrey) [inline appyx-components/stable/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/stable/backstack/BackStack.kt]: I wonder if we actually need stable in the package name? `backstack`, `testdrive`, etc are unique anyway

Comment (zsoltk) [reply appyx-components/stable/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/stable/backstack/BackStack.kt]: +1, I'd rather not have `stable` in the package name. (Let's keep it for the experimental and internal ones though!)

File: settings.gradle.kts [hunk 2]
```diff
@@ -26,15 +26,19 @@ dependencyResolutionManagement {
...
-    ":appyx-components:internal:android",
-    ":appyx-components:internal:common",
-    ":appyx-components:spotlight:android",
-    ":appyx-components:spotlight:common",
+    ":appyx-components:stable:backstack:android",
+    ":appyx-components:stable:backstack:common",
+    ":appyx-components:internal:test-drive:android",
+    ":appyx-components:internal:test-drive:common",
+    ":appyx-components:stable:spotlight:android",
+    ":appyx-components:stable:spotlight:common",
```
Comment (zsoltk) [inline settings.gradle.kts]: Let's group the stable ones together.
```suggestion
    ":appyx-components:internal:test-drive:android",
    ":appyx-components:internal:test-drive:common",
    ":appyx-components:stable:backstack:android",
    ":appyx-components:stable:backstack:common",
    ":appyx-components:stable:spotlight:android",
    ":appyx-components:stable:spotlight:common",
```

File: appyx-components/internal/test-drive/common/src/commonMain/kotlin/com/bumble/appyx/components/testdrive/TestDrive.kt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package com.bumble.appyx.components.internal.testdrive
```
Comment (zsoltk) [inline appyx-components/internal/test-drive/common/src/commonMain/kotlin/com/bumble/appyx/components/testdrive/TestDrive.kt]: Not a biggie, but: 

https://github.com/bumble-tech/appyx/pull/463#discussion_r1230939600

> Let's keep it for the experimental and internal ones though

**Pull request merged**